command/agent/secure_variable_endpoint.go

```diff
@@ -78,20 +80,30 @@ func (s *HTTPServer) secureVariableUpsert(resp http.ResponseWriter, req *http.Re
 	if SecureVariable.Items == nil {
 		return nil, CodedError(http.StatusBadRequest, "Secure variable missing required Items object.")
 	}
+
 	SecureVariable.Path = path
 
-	// Format the request
 	args := structs.SecureVariablesUpsertRequest{
 		Data: []*structs.SecureVariableDecrypted{&SecureVariable},
 	}
 	s.parseWriteRequest(req, &args.WriteRequest)
+	if err := parseCAS(req, &args); err != nil {
+		return nil, err
+	}
 
 	var out structs.SecureVariablesUpsertResponse
 	if err := s.agent.RPC(structs.SecureVariablesUpsertRPCMethod, &args, &out); err != nil {
+		if strings.Contains(err.Error(), "check-and-set conflict") {
+			q, _ := s.secureVariableQuery(resp, req, path)
+			sv := q.(*structs.SecureVariableDecrypted)
+			out.Conflict = make([]*structs.SecureVariableDecrypted, 1)
+			out.Conflict[0] = sv
+			resp.WriteHeader(http.StatusConflict)
+			return sv, nil
+		}
```

Comment:
`*structs.SecureVariableDecrypted` is available to the RPC. I think we can populate this `Conflict` field in the RPC handler and then let `len(out.Conflict)` determine whether we have to set the header. That'll let us avoid having to parse the error string or re-parse the original request.